ui-frontend/packages/catalog-ui-search/src/main/webapp/js/model/Query.js

```diff
@@ -335,18 +340,25 @@ Query.Model = Backbone.AssociatedModel.extend({
     data.batchId = Common.generateUUID()
 
     // Data.sources is set in `buildSearchData` based on which sources you have selected.
-    const sources = data.sources
+    let selectedSources = data.sources
+    const harvestedSources = Sources.getHarvested()
+
+    const isHarvested = (id) => harvestedSources.includes(id)
+    const isFederated = (id) => !harvestedSources.includes(id)
```

Comment:
Could `id` ever be the local catalog, in which case it isn't harvested or federated?

Reply:
"Harvested" in this scenario always includes the local catalog. The language is a bit confusing but if you check out the getHarvested() function it will always include the local catalog.